### PR TITLE
Simplify the logic of do_output_statistics.

### PR DESCRIPTION
**Pull request by bangerth**
First step towards addressing the issue mentioned yesterday on the forum by @maxrudolph . This just simplifies the logic a bit and will allow me, in the next step, access more member variables of the main class in the place where we actually do the writing.

### CODE REVIEW

File: source/simulator/helper_functions.cc [hunk 1]
```diff
@@ -189,43 +189,6 @@ namespace aspect
...
-     * of this function.
-     */
-    // We need to pass the arguments by value, as this function can be called on a separate thread:
-    void do_output_statistics (const std::string stat_file_name, //NOLINT(performance-unnecessary-value-param)
-                               const TableHandler *copy_of_table)
-    {
-      // write into a temporary file for now so that we don't
-      // interrupt anyone who might want to look at the real
-      // statistics file while the program is still running
-      const std::string tmp_file_name = stat_file_name + " tmp";
```
Comment (bangerth) [inline source/simulator/helper_functions.cc]: Not sure if this was intentional, but it should probably have been `".tmp"` (with a period) all along. Fixed in the patch while there.

Comment (tjhei) [reply source/simulator/helper_functions.cc]: looks like it, yes.

Comment (tjhei) [reply source/simulator/helper_functions.cc]: looks like it, yes.

**Pull request merged**